profiler/profile.go

```diff
@@ -236,6 +264,7 @@ func (t ProfileType) Tag() string {
 type profile struct {
 	// name indicates profile type and format (e.g. cpu.pprof, metrics.json)
 	name string
+	pt   ProfileType
```

Comment:
I added this for checking when a batch of profiles contains an execution trace, as opposed to checking the name.